### PR TITLE
Master hr presence arj

### PR DESCRIPTION
**Pull request by arj-odoo**
Make the employee presence icons more descriptive.
Allow to quickly filter the Presence Reports based on the user status.

TaskID: 2246447

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_attendance/models/hr_employee.py [hunk 1]
```diff
@@ -20,18 +20,20 @@ class HrEmployeeBase(models.AbstractModel):
...
+    @api.depends('user_id.im_status', 'attendance_state')
     def _compute_presence_state(self):
         """
         Override to include checkin/checkout in the presence state
         Attendance has the second highest priority after login
         """
         super()._compute_presence_state()
-        employees = self.filtered(lambda employee: employee.hr_presence_state != 'present')
+        employees = self.filtered(lambda e: e.hr_presence_state != 'present')
+        working_now_list = self._get_employee_working_now()
```
Comment (jbm-odoo) [inline addons/hr_attendance/models/hr_employee.py:31]: As `_get_employee_working_now`seems to be time-consuming, it better to reduce number of records.
```suggestion
       employee_to_check_working = employee.filtered(lambda e: e.attendance_state == 'checked_out' and employee.hr_presence_state == 'to_define')
        working_now_list = employee_to_check_working._get_employee_working_now()
```

File: addons/hr_presence/models/hr_employee.py [hunk 2]
```diff
@@ -21,16 +21,18 @@ class Employee(models.AbstractModel):
...
-        ('to_define', 'To Define')])
+        ('1_to_define', 'Absence To Check'),
+        ('2_absent', 'Justified Absence'),
+        ('3_present', 'Present')])
 
     def _compute_presence_state(self):
         super()._compute_presence_state()
         employees = self.filtered(lambda employee: employee.hr_presence_state != 'present' and not employee.is_absent)
         company = self.env.company
+        working_now_list = self._get_employee_working_now()
```
Comment (jbm-odoo) [inline addons/hr_presence/models/hr_employee.py:32]: Same here

File: addons/hr/models/hr_employee_base.py [hunk 3]
```diff
@@ -54,12 +57,13 @@ def _compute_presence_state(self):
         """
         # Check on login
         check_login = literal_eval(self.env['ir.config_parameter'].sudo().get_param('hr.hr_presence_control_login', 'False'))
+        working_now_list = self._get_employee_working_now()
```
Comment (jbm-odoo) [inline addons/hr/models/hr_employee_base.py:60]: Same here

File: addons/hr_presence/models/hr_employee.py [hunk 4]
```diff
@@ -21,16 +21,18 @@ class Employee(models.AbstractModel):
     # Stored field used in the presence kanban reporting view
     # to allow group by state.
     hr_presence_state_display = fields.Selection([
-        ('present', 'Present'),
-        ('absent', 'Absent'),
-        ('to_define', 'To Define')])
+        ('1_to_define', 'Absence To Check'),
+        ('2_absent', 'Justified Absence'),
+        ('3_present', 'Present')])
```
Comment (jbm-odoo) [inline addons/hr_presence/models/hr_employee.py:26]: Why did you change the technical terms?

File: addons/hr_holidays/views/hr_views.xml [hunk 5]
```diff
@@ -219,11 +273,17 @@
...
                 <field name="is_absent" invisible="1"/>
+                <field name="hr_icon_display" invisible="1"/>
                 <button name="%(hr_leave_action_new_request)d" type="action"
                         class="oe_stat_button"
                         invisible="context.get('from_my_profile', False)"
                         attrs="{'invisible': [('is_absent', '=', False)]}">
-                    <div role="img" class="fa fa-fw fa-plane o_button_icon text-danger" aria-label="Off Till" title="Off Till"/>
+                        <div attrs="{'invisible': [('hr_icon_display', '!=', 'presence_holiday_present')]}"
+                             role="img" class="fa fa-fw fa-plane o_button_icon text-success" aria-label="Off Till"
+                             title="Off Till"/>>
```
Comment (jbm-odoo) [inline addons/hr_holidays/views/hr_views.xml:283]:
```suggestion
                             title="Off Till"/>
```

File: addons/hr_attendance/models/res_config_settings.py [hunk 6]
```diff
@@ -8,4 +8,44 @@ class ResConfigSettings(models.TransientModel):
...
+        res = super(ResConfigSettings, self).execute()
+        return res
+
+    def plop(self):
+        # en temps normal
+        # False installed ir.module.module(163, )  ir.module.module()   ir.module.module()
+        a = 0
+        # Il faut checker si le module va être désinstallé et si le pin existe
+        # self.module_hr_attendance est false quand on décoche
+#         True quand c'est coché
```
Comment (jbm-odoo) [inline addons/hr_attendance/models/res_config_settings.py:50]: To remove?

File: addons/hr/models/hr_employee_base.py [hunk 7]
```diff
@@ -114,3 +118,43 @@ def _compute_address_id(self):
...
         for employee in self.filtered('department_id.manager_id'):
             employee.parent_id = employee.department_id.manager_id
+
+    @api.depends('resource_calendar_id', 'hr_presence_state')
+    def _compute_presence_icon(self):
+        """
+        This method compute the state defining the display icon in the kanban view.
+        It can be overriden to add other possibilities, like time off or attendances recordings.
+        """
+        # FIXME COMMENTS HERE
```
Comment (jbm-odoo) [inline addons/hr/models/hr_employee_base.py:128]: To remove?

File: addons/hr_presence/models/hr_employee.py (hunk 4 above)
Comment (arj-odoo) [reply addons/hr_presence/models/hr_employee.py:26]: The spec asked for this particular order and the prefix ensures the ordering in the view.

**Comment by arj-odoo** [comment]
@jbm-odoo I updated the _get_employee_working_now method to avoid unnecessary calculation.

**Comment by arj-odoo** [comment]
@tivisse After discussion, with @jbm-odoo We think we could use a selection field and selection_add in the overrides to limit the possibilities of the icons. What do you think?

File: addons/hr_presence/models/hr_employee.py (hunk 4 above)
Comment (tivisse) [reply addons/hr_presence/models/hr_employee.py:26]: I wouldn't change it. This requires a migration script. 

Reordering the keys should change the order for new databases (to confirm but I think that's the case), and old database remain that way.

**Comment by tivisse** [comment]
@robodoo r+
